### ISSUE TITLE
Add SSL to fedora and solr connections

### ISSUE THREAD

**Issue by kilahimm** [gwu-libraries/scholarspace#63]
Add SSL to tomcat server to secure connections.

**Comment by kilahimm** [comment]
Requested certificates and DNS entries for all test and prod servers to test and validate.

**Comment by kilahimm** [comment]
Still looking into this, currently encountering stack-trace errors on file-downloads and thumbnails fail to render when connecting over SSL to Fedora.

**Comment by kilahimm** [comment]
https://groups.google.com/forum/#!search/fedora$20ssl/hydra-tech/hN-nSrcKa-Q/NOYysQHyBQAJ

**Comment by kilahimm** [comment]
@kerchner let's coordinate on this one.  I can setup SSL on Tomcat, but it'll break the test site which is what I could use your help resolving.

**Comment by kilahimm** [comment]
With Fedora and Solr running on port 8443 and port 8080 redirected to port 8443 with basic auth, I'm able to successfully upload a new item into Fedora through Sufia.  The content file and the thumbnail are both properly created.

**Comment by kerchner** [comment]
👍

**Comment by kilahimm** [comment]
https://scholarspace-test.library.gwu.edu/downloads/9s1616164
and
https://scholarspace-test.library.gwu.edu/downloads/9s1616164?file=thumbnail

URLs still fail.

**Comment by kilahimm** [comment]
@kerchner I could use a fresh set of eyes on this.  I'm going to scope it into the next iteration potentially.

**Comment by kilahimm** [comment]
I've confirmed that the fedora logs are seeing the requests to for the thumbnails and content when using /download and in the search results.

**Comment by kilahimm** [comment]
@kerchner this commit to active-fedora is of interest: https://github.com/projecthydra/active_fedora/commit/05c7e1012321ca298800a424c44e117eca2207b9

**Comment by kilahimm** [comment]
So creating config/initializers/active_fedora.rb on GWSS with the content of the commit above resolves the issue.  active-fedora is constrained on Sufia 6.6.0 so we can't upgrade it to a release that includes the patched file.

**Comment by kerchner** [comment]
Outstanding. Good find.

**Comment by kerchner** [comment]
@kilahimm I've set this up on an AWS instance running Solr/Fedora to the extent that in a browser, I can accept (and _except_) the self-signed certs on the AWS server and successfully connect to both services using SSL.   The GWSS app, however, won't validate the self-signed cert
`Faraday::SSLError (SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed):`
as expected.

While I could go through the process of getting a GW DNS name and cert for the AWS server, I think we have ironed out the wrinkles with this in PR #228 - we've discovered and fixed the tomcat config files, added the needed steps into the README, etc.   We have SSL enabled on repository-test.library.gwu.edu and I can leave SSL off in my dev environment.  If needed, I can get a GW DNS and cert for the dev server.  So I am okay with closing this issue with the merge of PR #228.
